scripts/setup-macos.sh

```diff
@@ -29,16 +29,13 @@ set -e # Exit on error.
 set -x # Print commands that are executed.
 
 SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
-source $SCRIPTDIR/setup-helper-functions.sh
+source $SCRIPTDIR/setup-common.sh
 PYTHON_VENV=${PYHTON_VENV:-"${SCRIPTDIR}/../.venv"}
 
-NPROC=$(getconf _NPROCESSORS_ONLN)
-
-DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
-MACOS_VELOX_DEPS="bison boost double-conversion flex fmt gflags glog googletest icu4c libevent libsodium lz4 lzo openssl protobuf@21 simdjson snappy thrift xz xsimd zstd"
+MACOS_VELOX_DEPS="bison double-conversion flex gflags glog googletest icu4c libevent libsodium lz4 lzo openssl protobuf@21 simdjson snappy thrift xz zstd"
```

Comment:
Remove `double-conversion`. It is installed using the install function.

Reply:
@majetideepak I tried to resolve this dual install.
The problem with removing it here is that this is needed for boost. And the CI pipleine has a path where the actual dependencies are not installed. Only the brew dependencies are installed and the rest comes from bundling. 
double-conversion is not bundled and so if the setup script does not run the installation (remember we only turned it on for maOS 13) of the deps it'll fail.

So a couple of options:
1. Add it to be bundled
2. Remove install from being called and use version from brew (this just like with other dependencies means it can be updated at any time and cause issues).
3. Have the CI pipeline always run the setup script for macOS.

@assignUser What do you think?

Reply:
We could run the install scripts and cache the dependencies, that would give the best control over versions and have the smallest impact on ci times.

Reply:
Or we can do that later and just get the scripts in for now and optimize later?

Reply:
with the cache or stash action (though for this cache actually makes more sense) but yeah it's something we have to change in the workflow not the script, this PR is big enough as is we can do it as a follow up!

Reply:
Let me then build the dependencies always. There is a current problem with fmt due to it being upgraded in homebrew (because ccache upgrade). We want to make changes to use the predefined and installed versions instead of using uncontrolled build deps from homebrew (which will always be found first because of the linkage). So we don't want to just get the macOS deps from homebrew. Lets see what happens. 
This affects this PR as well.